examples/plot_metricframe_beyond_binary_classification.py

```diff
@@ -0,0 +1,245 @@
+    assert boxB[3] >= 0, "Bad delta y for boxB"
+
+    # Convert deltas to co-ordinates
+    boxA[2:4] = boxA[0:2] + boxA[2:4]
+    boxB[2:4] = boxB[0:2] + boxB[2:4]
+
+    # Determine the (x, y)-coordinates of the intersection rectangle
+    xA = max(boxA[0], boxB[0])
+    yA = max(boxA[1], boxB[1])
+    xB = min(boxA[2], boxB[2])
+    yB = min(boxA[3], boxB[3])
+
+    if (xB < xA) or (yB < yA):
+        return 0
+
+    # Compute the area of intersection rectangle
+    interArea = (xB - xA) * (yB - yA)
+
+    # Compute the area of both the prediction and ground-truth
+    # rectangles
+    boxAArea = (boxA[2] - boxA[0]) * (boxA[3] - boxA[1])
+    boxBArea = (boxB[2] - boxB[0]) * (boxB[3] - boxB[1])
+
+    # Compute the intersection over union by taking the intersection
+    # area and dividing it by the sum of prediction + ground-truth
+    # areas - the intersection area
+    iou = interArea / float(boxAArea + boxBArea - interArea)
+
+    return iou
+
+# %%
+# This is a metric for two bounding boxes, but for :class:`~fairlearn.metrics.MetricFrame`
+# we need to compare two lists of bounding boxes. For the sake of
+# simplicity, we will return the mean value of 'iou' for the
+# two lists, but this is by no means the only choice:
+
+
+def mean_iou(true_boxes, predicted_boxes):
+    assert len(true_boxes) == len(predicted_boxes), "Array size mismatch"
+
+    all_iou = [
+        bounding_box_iou(true_boxes[i], predicted_boxes[i])
+        for i in range(len(true_boxes))
+    ]
+
+    return np.mean(all_iou)
+
+# %%
+# We need to generate some input data, so first create a function to
+# generate a single random bounding box:
+
+
+def generate_bounding_box(max_coord, max_delta):
+    rng = np.random.default_rng()
+
+    corner = max_coord * rng.random(size=2)
+    delta = max_delta * rng.random(size=2)
+
+    return np.concatenate((corner, delta))
+
+# %%
+# Now use this to create sample `y_true` and `y_pred` arrays of
+# bounding boxes:
+
+
+def many_bounding_boxes(n_rows, max_coord, max_delta):
+    return [
+        generate_bounding_box(max_coord, max_delta)
+        for _ in range(n_rows)
+    ]
+
+
+true_bounding_boxes = many_bounding_boxes(n_rows, 5, 10)
+pred_bounding_boxes = many_bounding_boxes(n_rows, 5, 10)
+
+# %%
+# Finally, we can use these in a :class:`~fairlearn.metrics.MetricFrame`:
+
+mf_bb = MetricFrame(metrics={'mean_iou': mean_iou},
+                    y_true=true_bounding_boxes,
+                    y_pred=pred_bounding_boxes,
+                    sensitive_features=s_f)
+
+print("Overall metric")
+print(mf_bb.overall)
+print("Metrics by group")
+print(mf_bb.by_group)
+
+# %%
+# The individual entries in the `y_true` and `y_pred` arrays
+# can be arbitrarily complex. It is the metric functions
+# which give meaning to them. Similarly,
+# :class:`~fairlearn.metrics.MetricFrame` does not impose
+# restrictions on the return type. One can envisage an image
+# recognition task where there are multiple faces in each
```

Comment:
How about something that isn't face recognition? 🙂

Reply:
I can change to 'objects of interest' rather than 'faces' per se.... I don't have a handy other example. I'm sure there are plenty, but I'm hoping no one reading this would come away thinking "MetricFrame is only for binary classification _or_ image recognition scenarios?"

Reply:
I've added NLP as another example.

Reply:
Excellent! Yeah face recognition is just a bit loaded due to the many harmful applications. I'm all for assessing fairness of those obviously, but we wouldn't want to give the impression that we're endorsing such use cases. There are very good reasons why there's so much opposition... this is just one article but there are tons depending on how much time you'd like to spend on this: https://venturebeat.com/2020/06/10/amazon-imposes-one-year-moratorium-on-police-use-of-its-facial-recognition-technology/